src/main/java/org/medicmobile/webapp/mobile/FilePickerHandler.java

```diff
@@ -0,0 +1,163 @@
+	private Uri tempFileUri;
+
+	public FilePickerHandler(Activity context) {
+		this.context = context;
+	}
+
+	void setFilePickerCallback(ValueCallback<Uri[]> filePickerCallback) {
+		this.filePickerCallback = filePickerCallback;
+	}
+
+	void setTempFileUri(Uri tempFileUri) {
+		this.tempFileUri = tempFileUri;
+	}
+
+	void openPicker(FileChooserParams fileChooserParams, ValueCallback<Uri[]> filePickerCallback) {
+		trace(this, "FilePickerHandler :: Start file capture activity");
+		setFilePickerCallback(filePickerCallback);
+		setTempFileUri(null);
+		startFileCaptureActivity(fileChooserParams);
+	}
+
+	void processResult(int resultCode, Intent intent) {
+		if (resultCode != RESULT_OK) {
+			String message = "FilePickerHandler :: Bad result code: %s. " +
+				"Either the user didn't select a file or there's an error during the operation.";
+			warn(this, message, resultCode);
+			sendDataToWebapp(null);
+			return;
+		}
+
+		sendDataToWebapp(intent);
+	}
+
+//> PRIVATE
+
+	/**
+	 * Executes the file picker callback.
+	 * The callback will give control back to WebView that automatically fetches the file by using a
+	 * provided URI, finally it assigns the bytes to the form's input. Even if there's not URI (ie null),
+	 * it needs to give control back to WebView, otherwise when clicking again on the form's input
+	 * it won't open the file picker.
+	 *
+	 * @param intent Intent from activity's response.
+	 */
+	private void sendDataToWebapp(Intent intent) {
+		if (this.filePickerCallback == null) {
+			warn(this, "FilePickerHandler :: Cannot send data back to webapp, filePickerCallback is null.");
+			return;
+		}
+
+		Uri uri = getSelectedFileUri(intent);
+		trace(this, "FilePickerHandler :: Sending data back to webapp, URI: %s", uri);
+		this.filePickerCallback.onReceiveValue(uri == null ? null : new Uri[] { uri });
+		this.filePickerCallback = null;
+	}
+
+	private Uri getSelectedFileUri(Intent intent) {
+		Uri uri = intent == null ? null : intent.getData();
+
+		if (uri == null && this.tempFileUri != null) {
+			uri = this.tempFileUri;
+			setTempFileUri(null);
+		}
+
+		return uri;
+	}
+
+	private void startFileCaptureActivity(FileChooserParams fileChooserParams) {
+		String[] mimeTypes = fileChooserParams.getAcceptTypes();
+		if (mimeTypes == null) {
+			warn(this, "FilePickerHandler :: MIME type is null, please specify a MIME type.");
+			sendDataToWebapp(null);
+			return;
+		}
+
+		trace(this, "FilePickerHandler :: Accepted MIME types: %s", Arrays.toString(mimeTypes));
+
+		Intent pickerIntent = new Intent(Intent.ACTION_GET_CONTENT);
+		pickerIntent.addCategory(Intent.CATEGORY_OPENABLE);
+		pickerIntent.putExtra(Intent.EXTRA_LOCAL_ONLY, true);
+		Intent captureIntent = null;
+
+		if (mimeTypes.length == 1) {
+			String mimeType = mimeTypes[0];
+			pickerIntent.setType(mimeType);
+			captureIntent = getCaptureIntent(mimeType);
+		} else {
+			pickerIntent.setType("*/*");
+			pickerIntent.putExtra(Intent.EXTRA_MIME_TYPES, mimeTypes);
```

Comment:
I'm not sure how this should work. I created a form with a binary input and an empty media type, and I got: 
![Screenshot_20220221-102944](https://user-images.githubusercontent.com/35681649/154917175-f8dba056-b9eb-496a-9577-63cbd59750a8.png)

Reply:
Can you please the xml for this input? Thanks!

Reply:
```
<bind nodeset="/photos-on-android/my-any-file" type="binary"/>
....
<upload ref="/photos-on-android/my-any-file" mediatype="">
			<hint>Select a whatever?</hint>
			<label>File widget</label>
		</upload>
```

Reply:
Done

Reply:
The behaviour is: if there isn't a mime type, it shouldn't let to select a file.

Reply:
Ok, so right now, if I click on this field, nothing should happen, right?